### PR TITLE
Geopoint fused location re-activated

### LINKED ISSUE #2717: Switch all location capture to FusedLocationProvider
- [x] Implement changes
- [ ] Explore and document how users can force GPS-only results
- [ ] Understand and document implications for external high-precision GPS sensors
- [ ] Document implications for accuracy and location data availability for user-facing docs

`FusedLocationProvider` is the location provider recommended by Google. It's supposed to balance accuracy, resource usage, etc.

Collect used to use it but in Feb 2018, it started capping accuracy at 10m (https://github.com/opendatakit/collect/issues/2008). This turned out to be [an Android bug](https://issuetracker.google.com/u/0/issues/79189573#comment20) which is now fixed. However, accuracy **[is still capped at 3m](https://issuetracker.google.com/issues/118789585)**.

It would probably make sense for the 3m issue to be addressed before switching back to using it.

Regardless, there are some higher-level questions about whether it's the right way to go. It's supposed to provide better battery management but that's only if less accuracy is acceptable. It may make sense to continue only using the GPS since accuracy is almost always important. That said, there are certain environments (urban) where other location sources may be more accurate. Users could force using the GPS by turning other providers off at the Android level, I think, but that may be harder to use.

This may be a conversation more appropriate for the forum. I wanted to at least file the issue so the content and history is not forgotten but feel free to close and start a thread elsewhere or comment here.

### PR DESCRIPTION
**Pull request by boscacci**
Closes #2717 

#### Before submitting this PR, please make sure you have:
- [ ] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [ ] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [ ] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by grzesiek2010** [comment]
Hi @boscacci 

congrats on your first pr :pizza: :pizza: :pizza: 
Please avoid unnecessary changes like in this case in file `build.gradle` and `gradle/wrapper/gradle-wrapper.properties`. Could you remove them?

File: collect_app/src/main/java/org/odk/collect/android/location/client/LocationClients.java [hunk 1]
```diff
@@ -27,14 +27,18 @@ private LocationClients() {
      * 3/14/2018 - GoogleLocationClient removed because of user reports that accuracy does not get
      * better than 10m.
      *
+     * 1/3/2019 - GoogleLocationClient implemented because accuracy has improved to 3m
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/location/client/LocationClients.java:30]: Please also remove this change. Adding comments every time we edit a file is not our practice. Everyone who wants to find out why you added this change can just use git, read the issue/ pr /description of your commits and that's enough.

Comment (boscacci) [reply collect_app/src/main/java/org/odk/collect/android/location/client/LocationClients.java:30]: Giving it a shot now!

**Comment by grzesiek2010** [comment]
Closed in favor of #2793

**Pull request closed without merging**